tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.NetFramework.cs

```diff
@@ -37,9 +37,16 @@ private static Assembly AssemblyResolve_ManagedProfilerDependencies(object sende
             }
 
             var path = Path.Combine(ManagedProfilerDirectory, $"{assemblyName}.dll");
+
             if (File.Exists(path))
             {
-                StartupLogger.Debug("Loading {0}", path);
+                if (args.Name.StartsWith("Datadog.Trace, Version=") && args.Name != AssemblyName)
+                {
+                    StartupLogger.Debug("Trying to load {0} which does not match the expected version ({1})", args.Name, AssemblyName);
+                    return null;
```

Comment:
Is this case purely hypothetical or did it solve an issue that occurred in your testing? I'm trying to understand this change a little better

Reply:
It's a precaution but now I'm not sure it's actually needed.

Basically, we first had an error in the profiler code where we were binding the DistributedTracer calls to the wrong version of the assembly, yet everything was working. It turns out it worked because the loader was redirecting the load regardless of the version. After this change, is we mess up the version in the profiler then it will fail with this error:

```
 Unhandled Exception: System.TypeInitializationException: The type initializer for 'Datadog.Trace.ClrProfiler.DistributedTracer' threw an exception. ---> System.IO.FileLoadException: Could not load file or assembly 'Datadog.Trace, Version=9.99.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
    at Datadog.Trace.ClrProfiler.DistributedTracer.GetDistributedTracer()
    at Datadog.Trace.ClrProfiler.DistributedTracer..cctor()
    --- End of inner exception stack trace ---
    at Datadog.Trace.ClrProfiler.DistributedTracer.get_Instance()
    at Datadog.Trace.Tracer.CreateSpanContext(ISpanContext parent, String serviceName, Boolean ignoreActiveScope, Nullable`1 traceId, Nullable`1 spanId)
    at Datadog.Trace.Tracer.StartSpan(String operationName, ITags tags, ISpanContext parent, String serviceName, Nullable`1 startTime, Boolean ignoreActiveScope, Nullable`1 traceId, Nullable`1 spanId, Boolean addToTraceContext)
    at Samples.VersionConflict_2x.Program.<Main>d__0.MoveNext()
 --- End of stack trace from previous location where exception was thrown ---
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
    at Samples.VersionConflict_2x.Program.<Main>()
```

But do we really need to fail if we get the version wrong, I don't know. It kind of feel safer because it means we will catch it during the tests.

Reply:
Makes sense, if it will help increase visibility in the tests then let's keep it 👍🏼